homeassistant/components/maxcube/binary_sensor.py

```diff
@@ -20,36 +22,47 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         add_entities(devices)
 
 
-class MaxCubeShutter(BinarySensorEntity):
-    """Representation of a MAX! Cube Binary Sensor device."""
-
+class MaxCubeBinarySensorBase(BinarySensorEntity):
     def __init__(self, handler, device):
         """Initialize MAX! Cube BinarySensorEntity."""
-        room = handler.cube.room_by_id(device.room_id)
-        self._name = f"{room.name} {device.name}"
         self._cubehandle = handler
         self._device = device
+        self._room = handler.cube.room_by_id(device.room_id)
 
-    @property
-    def name(self):
-        """Return the name of the BinarySensorEntity."""
-        return self._name
+    def update(self):
+        """Get latest data from MAX! Cube."""
+        self._cubehandle.update()
 
-    @property
-    def unique_id(self):
-        """Return a unique ID."""
-        return self._device.serial
 
-    @property
-    def device_class(self):
-        """Return the class of this sensor."""
-        return DEVICE_CLASS_WINDOW
+class MaxCubeShutter(MaxCubeBinarySensorBase):
+    """Representation of a MAX! Cube Binary Sensor device."""
+
+    def __init__(self, handler, device):
+        """Initialize MAX! Cube BinarySensorEntity."""
+        super().__init__(handler, device)
+
+        self._attr_name = f"{self._room.name} {self._device.name}"
+        self._attr_unique_id = self._device.serial
+        self._attr_device_class = DEVICE_CLASS_WINDOW
 
     @property
     def is_on(self):
         """Return true if the binary sensor is on/open."""
         return self._device.is_open
 
-    def update(self):
-        """Get latest data from MAX! Cube."""
-        self._cubehandle.update()
+
+class MaxCubeBattery(MaxCubeBinarySensorBase):
+    """Representation of a MAX! Cube Binary Sensor device."""
+
+    def __init__(self, handler, device):
+        """Initialize MAX! Cube BinarySensorEntity."""
+        super().__init__(handler, device)
+
+        self._attr_name = f"{self._room.name} {device.name} battery"
+        self._attr_unique_id = f"{self._device.serial}_battery"
+        self._attr_device_class = DEVICE_CLASS_BATTERY
```

Comment:
We should add entity_category as well, as this is a diagnostic item.

Reply:
Set it to windows shutter entity and battery entity - 34689923b132854b32a6a940942feb4730e4b7ad

Reply:
Maybe I'm not looking correctly, but that commit isn't in this PR?

Reply:
@frenck it's in the base class `MaxCubeBinarySensorBase`

Reply:
@frenck, it's in PR, I did `fixup` commit, this commit is now squashed.

Reply:
Anyways, this integration would benefit from using EntityDescriptions, which removes the magic :)

Reply:
> Please, don't squash. Really, it makes review impossible

OK, will squash when review is done

> Anyways, this integration would benefit from using EntityDescriptions, which removes the magic :)

Explain, what is it, link?

Reply:
> Explain, what is it, link?

No link, but searching our codebase will give examples enough. Nevertheless, not for this PR.

Reply:
> OK, will squash when review is done

We squash on merge. Please don't squash yourself. Thanks 👍

Reply:
> We squash on merge. Please don't squash yourself. Thanks ￼

Do you squash all commits into one? I would keep 2 commits, one that refactors old code and another that adds new functionality.

> PS: Please use descriptive commit messages. "fixup!" isn't helpful.

`fixup!` commits are git way to fix previous commits - https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---fixupamendrewordltcommitgt

Reply:
@petslane if you want to keep 2 commits, then you need to provide two pull requests. Us contributors cannot influence how maintainers merge them in, and it is the policy of the HA maintainers to always merge them squashed.

If your PR is 100% polished on the first attempt, then just one commit is great. If it isn't polished, and recommendations are made, then as @frenck requested it is much easier for reviewers to follow-up if the comments are descriptive (eg. `fix pylint` or `fix black` or `fix test`).

Reply:
@epenet thanks for explaining. Looking at `dev` branch commits then looks like the most important is the PR title.

Reply:
Yes, Github automatically picks up the PR title as the merge message in case of multiple commits. If there's a single commit then Github seems to default back to the commit description which is annoying.

If you have the will once this is merged in, then you could register yourself as a codeowner for `maxcube`, and take a look at implementing `EntityDescription` for all the `maxcube` platforms.

See this for code-owners: https://github.com/home-assistant/core/blob/63c9cfdbc8f5335a133abe66d1ca1b48ff4be6da/CODEOWNERS#L300-L302

See this for EntityDescription: https://github.com/home-assistant/core/blob/63c9cfdbc8f5335a133abe66d1ca1b48ff4be6da/homeassistant/components/onewire/binary_sensor.py#L36-L38